### PR TITLE
docs: introducing APISIX's testing framework

### LINKED ISSUE #4534: apisix plugin write test cases
你好，请问apisix plugin 如何编写测试用例，有没有好的最佳实践？

### PR DESCRIPTION
**Pull request by spacewander**
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

Fix #4534 

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: docs/en/latest/internal/testing-framework.md [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+#
+-->
+
+APISIX uses a testing framework based on our fork of test-nginx: https://github.com/iresty/test-nginx.
+For details, you can check the documentation of this project.
+
+If you want to test the CLI behavior of APISIX (`./bin/apisix`),
+you need to write a shell script in the t/cli directory to test it. You can refer to the existing test scripts.
+
+If you want to test the others, you need to write test code based on the framework.
```
Comment (tokers) [inline docs/en/latest/internal/testing-framework.md:30]:
```suggestion
If you want to test the others, you need to write test codes based on the framework.
```

File: docs/en/latest/internal/testing-framework.md [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+APISIX uses a testing framework based on our fork of test-nginx: https://github.com/iresty/test-nginx.
+For details, you can check the documentation of this project.
+
+If you want to test the CLI behavior of APISIX (`./bin/apisix`),
+you need to write a shell script in the t/cli directory to test it. You can refer to the existing test scripts.
```
Comment (tokers) [inline docs/en/latest/internal/testing-framework.md:28]:
```suggestion
you need to write a shell script in the t/cli directory to test it. You can refer to the existing test scripts for more details.
```

File: docs/en/latest/internal/testing-framework.md [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+
+If you want to test the others, you need to write test code based on the framework.
+
+Here, we briefly describe how to do simple testing based on this framework.
+
+## Test file
+
+you need to write test cases in the t/ directory, in a corresponding `.t` file. Note that a single test file should not exceed 800 lines, and if it is too long, it needs to be divided by a suffix. For example:
```
Comment (tokers) [inline docs/en/latest/internal/testing-framework.md:36]:
```suggestion
you need to write test cases in the t/ directory, in a corresponding `.t` file. Note that a single test file should not exceed `800` lines, and if it is too long, it needs to be divided by a suffix. For example:
```

File: docs/en/latest/internal/testing-framework.md [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+chash_key: "custom-one"
+hash_on: header
+chash_key: "custom-one"
+hash_on: header
+chash_key: "custom-one"
+hash_on: header
+chash_key: "custom-one"
+```
+
+The default log level is info, but you can get the debug level log with `-- log_level: debug`.
```
Comment (tokers) [inline docs/en/latest/internal/testing-framework.md:224]:
```suggestion
The default log level is `info`, but you can get the debug level log with `-- log_level: debug`.
```

File: docs/en/latest/internal/testing-framework.md (hunk 1 above)
Comment (spacewander) [reply docs/en/latest/internal/testing-framework.md:30]: "code" used by the computer is uncountale

**Review by tokers — APPROVED** [review]

File: docs/en/latest/internal/testing-framework.md [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+### Debugging test cases
+
+The Nginx configuration and logs generated by the test cases are located in the t/servroot directory. The Nginx configuration template for testing is located in t/APISIX.pm.
+
+### Running only some test cases
+
+Three notes can be used to control which parts of the tests are executed.
+
+```
+--- FIRST
```
Comment (tzssangglass) [inline docs/en/latest/internal/testing-framework.md:259]: Can add a command line to show how to use it

**Review by tzssangglass — APPROVED** [review]

**Pull request merged**